cmd/helm/root.go

```diff
@@ -430,5 +433,77 @@ func newRootCmd(actionConfig *action.Configuration, out io.Writer, args []string
 	// Find and add plugins
 	loadPlugins(cmd, out)
 
+	// Add template functions which will be used in the usage template
+	cobra.AddTemplateFunc("LoggingFlags", LoggingFlags)
+	cobra.AddTemplateFunc("CapitalizeFlagDescriptions", CapitalizeFlagDescriptions)
+	cmd.SetUsageTemplate(CmdUsageTemplate)
 	return cmd
 }
+
+// LoggingFlags will go through all the logging flags which have the annotation: "log":"flag" and
+// get the appropriate usage
+func LoggingFlags(cmd *cobra.Command) string {
+
+	globalFlags := cmd.Flags()
+	loggingFlags := &pflag.FlagSet{}
+
+	// Retrieve all the flags that have that particular annotation
+	globalFlags.VisitAll(func(f *pflag.Flag) {
+		if reflect.DeepEqual(f.Annotations["log"], []string{"flag"}) {
+			loggingFlags.AddFlag(f)
+		}
+	})
+
+	// Weird hack with cobra13. Essentially, cobra doesn't have a function to return a non-pointer-set of flags
+	// so we'll set the flags as non-hidden, get the descriptions, then set them back to hidden
```

Comment:
is there a bug that should be referenced here? That way we can track this issue in the future and remove this hack once it's fixed upstream.